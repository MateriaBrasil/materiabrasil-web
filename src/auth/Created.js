import React, { Component } from 'react';
import Redirect from 'react-router-dom/Redirect';
import ReactGA from 'react-ga';

export default class extends Component {
  constructor(props) {
    super(props);
    const { created, setCurrentUser } = props;

    ReactGA.event({
      category: 'Cadastro Usu√°rio',
      action: 'Submit',
    });

    setCurrentUser(created);
  }

  render() {
    const { location } = this.props;
    const { state } = location;
    const path = (state && state.referrer) || '/';

    return <Redirect to={path} />;
  }
}
