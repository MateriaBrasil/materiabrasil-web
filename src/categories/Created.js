import { Component } from 'react'
import find from 'lodash/find'
import map from 'lodash/map'
import filter from 'lodash/filter'
import isEmpty from 'lodash/isEmpty'
import flatten from 'lodash/flatten'

const findCategory = (categories, categoryId) => {
  const category = find(categories, ({ id }) => id === categoryId)

  if (category) {
    return category
  }

  const children = flatten(map(categories, category => category.children))
  return isEmpty(children) ? null : findCategory(children, categoryId)
}

export default class extends Component {
  constructor(props) {
    super(props)

    const { setMaterialCategories, materialCategories, categories } = props
    const { created } = props

    const createdCategory = findCategory(categories, created.categoryId)
    const parent = findCategory(categories, createdCategory.parentId)

    if (parent && parent.multipleChoice) {
      return
    }

    setMaterialCategories(
      filter(materialCategories, ({ categoryId }) => {
        if (categoryId === created.categoryId) {
          return true
        }

        const category = findCategory(categories, categoryId)
        return category.parentId !== createdCategory.parentId
      }),
    )
  }

  render() {
    return null
  }
}
