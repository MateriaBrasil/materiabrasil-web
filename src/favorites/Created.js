import { Component } from 'react'

export default class extends Component {
  componentDidMount() {
    this.props.snackbar.actions.setMessage('Favorito adicionado com sucesso')
  }

  render() {
    return null
  }
}
