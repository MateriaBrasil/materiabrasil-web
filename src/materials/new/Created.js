import React, { Component } from 'react'
import Redirect from 'react-router-dom/Redirect'

export default class extends Component {
  componentDidMount() {
    this.props.snackbar.actions.setMessage('Material cadastrado com sucesso')
  }

  render() {
    return <Redirect to={`/materials/${this.props.material.id}`} />
  }
}
