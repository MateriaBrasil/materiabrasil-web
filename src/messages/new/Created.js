import React, { Component } from 'react';
import Redirect from 'react-router-dom/Redirect';

export default class extends Component {
  render() {
    return (
      <Redirect
        to={`/${this.props.toType.toLowerCase()}s/${
          this.props.match.params.id
        }/messages/success`}
      />
    );
  }
}
