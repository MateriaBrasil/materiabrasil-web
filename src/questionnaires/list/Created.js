import { Component } from 'react'
import find from 'lodash/find'

export default class extends Component {
  constructor(props) {
    super(props)

    const { setAnswers, questionnairesAnswers, created } = props

    const answer = find(
      questionnairesAnswers,
      ({ questionId }) => questionId === created.questionId,
    )

    answer.optionId = created.optionId

    setAnswers(questionnairesAnswers)
  }

  render() {
    return null
  }
}
