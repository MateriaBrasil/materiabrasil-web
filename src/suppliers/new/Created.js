import React, { Component } from 'react'
import Redirect from 'react-router-dom/Redirect'

export default class extends Component {
  constructor(props) {
    super(props)
    const { currentUser, supplier, setCurrentUser, snackbar } = props

    setCurrentUser({
      ...currentUser,
      suppliers: [...currentUser.suppliers, supplier],
    })

    snackbar.actions.setMessage('Fornecedor cadastrado com sucesso')
  }

  render() {
    const { supplier } = this.props
    return <Redirect to={`/suppliers/${supplier.id}?reload=true`} />
  }
}
