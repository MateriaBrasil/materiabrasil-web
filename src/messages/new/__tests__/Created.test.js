import React from 'react'
import renderer from 'react-test-renderer'

import Created from '../Created'

it('renders correctly', () => {
  const goBack = jest.fn()
  const props = { history: { goBack } }

  const tree = renderer.create(<Created {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(goBack).toHaveBeenCalledWith()
})
