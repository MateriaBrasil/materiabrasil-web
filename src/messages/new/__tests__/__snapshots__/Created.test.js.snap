// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  to="/users/3/messages/success"
>
  Redirect
</div>
`;
